packages/core/test/unit/query-generator/attributes-to-sql.test.ts

```diff
@@ -0,0 +1,221 @@
+      default: { id: 'INTEGER DEFAULT 0' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', defaultValue: undefined } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', defaultValue: undefined } }), {
+      default: { id: 'INTEGER' },
+      mssql: { id: 'INTEGER NULL' },
+    });
+  });
+
+  it('Add column level comment', () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', comment: 'Test' } }), {
+      default: { id: 'INTEGER COMMENT Test' },
+      'mariadb mysql snowflake': { id: `INTEGER COMMENT 'Test'` },
+      mssql: { id: 'INTEGER NULL COMMENT Test' },
+      'sqlite ibmi': { id: 'INTEGER' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true } }), {
+      default: { id: 'INTEGER UNIQUE' },
+      mssql: { id: 'INTEGER NULL UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      postgres: { id: `INTEGER UNIQUE; COMMENT ON COLUMN "foo"."bar"."column" IS 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', after: 'Bar' } }`, () => {
+    // @ts-expect-error -- after is not accepted in AttributeOptions so attributesToSQL might need to be typed differently
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', after: 'Bar' } }), {
+      default: { id: 'INTEGER' },
+      'mariadb mysql': { id: 'INTEGER AFTER `Bar`' },
+      mssql: { id: 'INTEGER NULL' },
+      'snowflake ibmi': { id: 'INTEGER AFTER "Bar"' },
+    });
+  });
+
+  // TODO: check what this test is supposed to do and update the possible defaultValue accordingly
```

Comment:
This test is broken and should be split into multiple tests:

- One that uses `DataTypes.ARRAY` (on dialects that support this type) with an empty array: should work
- One that uses `DataTypes.BLOB`: should never work, `[]` is not a valid value on a blob
- One that uses `DataTypes.JSON` (on dialects that support this type) : should always work 
- One that uses any string type (like `'JSON'` instead `DataTypes.JSON`): should never work because can't guess the type of an empty array and string types don't give any information

Note that the data type is an actual data type and not a string in the above two tests